toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -939,6 +939,52 @@ func InstallGrubEnv(installRoot string) (err error) {
 	return
 }
 
+// CopyAndChangeModeGrubDefaults copies a file from src to dst, creating directories with the given 
+// access rights for the destination if needed. dst is assumed to be a file and not a directory.
+// Will change the permissions to the given value.
+// In the case of handling grub defaults, we must additionally preserve the state of dst, appending it
+// to the src template before stomping. This is to preserve any configurations that might have been
+// installed by packages before the tools install our template. 
+// - src is the Mariner-tools-defined template file
+// - dst holds the root partition
+func CopyAndChangeModeGrubDefaults(src, dst string, dirmode os.FileMode, filemode os.FileMode) (err error) {
+	const (
+		tmpFile = "/installer/grub2/grub.tmp"
+	)
+	
+	// Transfer pre-existing /etc/default/grub configuration (e.g. from package installs)
+	// to a tmp file so that package-related settings are preserved when stomping dst with
+	// our template file. 
+	srcContent, err := file.ReadLines(src)
+	if err != nil {
+		return
+	}
+	dstContent, err := file.ReadLines(dst)
+	if err != nil {
+		return
+	}
+	_, stderr, err := shell.Execute("touch", tmpFile)
+	if err != nil {
+		logger.Log.Warnf("Error while creating tmp grub defaults file: %v", stderr)
+		return
+	}
+	err = file.AppendSeries(srcContent, tmpFile)
+	if err != nil {
+		return
+	}
+	err = file.AppendSeries(dstContent, tmpFile)
+	if err != nil {
+		return
+	}
+
+	err = file.CopyAndChangeMode(tmpFile, dst, dirmode, filemode)
+	if err != nil {
+		return
+	}
+	_, stderr, err = shell.Execute("rm", tmpFile)
+	return
+}
+
```

Comment:
i have no issues with the code here, i'm just constantly amazed by how much seemingly redundant and unnecessary code the go language requires to do simple things...